common/flatpak-dir.c

```diff
@@ -4637,6 +4670,14 @@ repo_pull_local_untrusted (FlatpakDir          *self,
         }
     }
 
+  repo_path = g_file_get_path (ostree_repo_get_path (repo));
+  if (stat (repo_path, &st_buf) == 0 &&
```

Comment:
Wouldn't it make sense for the system-helper to pass a boolean here to certify that the source repo is immutable, rather than checking the permissions? Technically you have no way of knowing at this point that there are no bindfs mounts allowing non-root write access. For example, a code path added in the future could conceivably arrive here without properly killing bindfs mounts.

Also technically the repo directory could have `0755` permissions and a file in the directory could have more open permissions, so a non-root user could still modify the contents.